src/com/google/javascript/jscomp/deps/ModuleResolver.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2017 The Closure Compiler Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.javascript.jscomp.deps;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.javascript.jscomp.ErrorHandler;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/** Base class for algorithms that resolve JavaScript module references to input files. */
+public abstract class ModuleResolver {
+  /** The set of all known input module URIs (including trailing .js), after normalization. */
+  protected final ImmutableSet<String> modulePaths;
+
+  /** Root URIs to match module roots against. */
+  protected final ImmutableList<String> moduleRootPaths;
+
+  protected final ErrorHandler errorHandler;
+
+  public ModuleResolver(
+      ImmutableSet<String> modulePaths,
+      ImmutableList<String> moduleRootPaths,
+      ErrorHandler errorHandler) {
+    this.modulePaths = modulePaths;
+    this.moduleRootPaths = moduleRootPaths;
+    this.errorHandler = errorHandler;
+  }
+
+  Map<String, String> getPackageJsonMainEntries() {
+    return null;
```

Comment:
Could we return an empty map instead of null so this method doesn't have to be `@Nullable`?